modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/capability/xml/schema/ClasspathBasedDslContext.java

```diff
@@ -45,7 +46,7 @@
   private final ClassLoader classLoader;
   private final Map<String, Class<?>> extensionsByName = new HashMap<>();
   private final Map<String, ExtensionModel> resolvedModels = new HashMap<>();
-  private TypeCatalog typeCatalog;
+  private LazyValue<TypeCatalog> typeCatalog = new LazyValue<>(() -> TypeCatalog.getDefault(getExtensions()));
```

Comment:
TypeCatalog will be filled with all the extensions declared types. DslContext will search any typeId and find the extension which is the owner (aka the one who contributed with) that type.